### ISSUE TITLE
microbit pxt - strange workflow with radio blocks

### ISSUE THREAD

**Issue by whaleygeek** [Microsoft/pxt#641]
microbit.org-ticket: 203
editor-ver: pxt.microbit.org as of 01 Nov 2016, 17:20 GMT

There is a strange workflow error appearing when using the radio, blocks - something to do with how the inner packet is configured. I have captured a workflow using screenshots as it is quite hard to describe in words, but it generates different code, which I've not been able to completely track down. 

I was wondering if someone could work through this workflow and see if they can spot what is going wrong? The parameter names and formats to the function change just by dragging the packet blocks out and back in again, suggesting that the default format when dropping down a radio block is different from the version generated from editing the packet format.

**Comment by whaleygeek** [comment]
P.S. Sorry I incorrectly put a showString in there - ignore that, will take screenshots again...

**Comment by whaleygeek** [comment]
Here is the workflow in new screenshots (the names of the parameters change when reconfiguring the data packet, and before they had a : pair, after they have a single name)...
![1](https://cloud.githubusercontent.com/assets/4086785/19899948/ce930958-a059-11e6-9f34-e87e70d281c2.png)
![2](https://cloud.githubusercontent.com/assets/4086785/19899953/d1e5a818-a059-11e6-90e5-f3e7e5044d81.png)
![3](https://cloud.githubusercontent.com/assets/4086785/19899957/d4ebf9a4-a059-11e6-9a44-d266d146faee.png)
![4](https://cloud.githubusercontent.com/assets/4086785/19899960/d6e9185e-a059-11e6-9314-166c7739ee15.png)
![5](https://cloud.githubusercontent.com/assets/4086785/19899964/d8e003f2-a059-11e6-839b-0294e3dd4e93.png)
![6](https://cloud.githubusercontent.com/assets/4086785/19899968/db918148-a059-11e6-84de-7bb90695cafe.png)
![7](https://cloud.githubusercontent.com/assets/4086785/19899972/dce2fb6c-a059-11e6-8dcf-3beb6d5e523d.png)
![8](https://cloud.githubusercontent.com/assets/4086785/19899974/def997a8-a059-11e6-980b-7764ac7f661a.png)
![9](https://cloud.githubusercontent.com/assets/4086785/19899978/e345f914-a059-11e6-977c-02ba4732b21d.png)
![10](https://cloud.githubusercontent.com/assets/4086785/19899981/e564c1f8-a059-11e6-9eb5-1c5f0224c4ae.png)

**Comment by abchatra** [comment]
@riknoll Can you summarize your understanding of this issue here?

**Comment by riknoll** [comment]
Basically, if you change the names of the variables being used in the packet and then use the mutator to remove them and re-add them, the changed names are not persisted. The issue with that behavior is that if you used the old names somewhere else in your program, those uses do not get changed.

The fix is to make it so that the names of the variables are persisted across changes in the mutators. I can do the fix, but it should wait for my other mutator stuff to be merged.

**Comment by whaleygeek** [comment]
Hello, thanks for tagging this as a P1 bug - do you know which release this fix is likely to be targeted at? Thanks!

**Comment by jaustin** [comment]
Do we have any update on the resolution of this? I guess it's waiting for "the other mutator stuff" - but we are still seeing bug reports that boil down to this issue.

**Comment by pelikhan** [comment]
@riknoll
